backend/internal/server/apis/sports/stats/penalty.go

```diff
@@ -55,11 +104,61 @@ func postPenaltyHandler(c *fiber.Ctx) error {
 	}
 
 	db := db.GetSession(c)
-	err = db.CreatePenalty(penaltyRequest)
+	penalty, err := db.CreatePenalty(penaltyRequest)
 	if err != nil {
 		log.WithErr(err).Alert("Failed to save penalty request")
 		return responder.InternalServerError(c)
 	}
 
+	return responder.OkWithData(c, penalty)
+}
+
+func putPenaltyHandler(c *fiber.Ctx) error {
+	id := c.Params("id")
+	log := locals.Logger(c)
+
+	// Parse penalty
+	penaltyRequest := &models.Penalty{}
+	err := c.BodyParser(penaltyRequest)
+	if err != nil {
+		log.WithErr(err).Alert("Failed to parse penalty request payload")
+		return responder.BadRequest(c, "Failed to parse penalty request payload")
+	}
+	if strconv.FormatUint(uint64(penaltyRequest.ID), 10) != id { // Ugly, looking for a better way to do this
+		log.WithErr(err).Alert("Penalty ID in URL does not match ID in payload")
+		return responder.BadRequest(c, "Penalty ID in URL does not match ID in payload")
+	}
+
+	// Update penalty
+	db := db.GetSession(c)
+	penalty, err := db.UpdatePenalty(penaltyRequest)
+	if err != nil {
+		log.WithErr(err).Alert("Failed to update penalty with ID %s", id)
+		return responder.InternalServerError(c)
+	}
+
+	return responder.OkWithData(c, penalty)
+}
+
+func deletePenaltyHandler(c *fiber.Ctx) error {
+	id := c.Params("id")
+	log := locals.Logger(c)
+	db := db.GetSession(c)
+
+	// Get penalty
+	penalty, err := db.GetPenaltyByID(id)
+	if err != nil {
+		log.WithErr(err).Alert("Delete penalty failed because there is no penalty with ID %s", id)         // TODO: Not working
```

Comment:
ℹ️  I don't think we want an alert here, I think error would be fine -- what isn't working?

Reply:
Fixed, good catch. Thanks